elk/utils_evaluation/ccs.py

```diff
@@ -1,57 +1,46 @@
-import numpy as np
+from typing import Literal
 import torch
 import matplotlib.pyplot as plt
+from elk.utils_evaluation.probes import Probe, OriginalLinearProbe
+from attrs import define
 
 
-class CCS(object):
+@define
+class TrainParams:
+    num_epochs: int = 1000
+    num_tries: int = 10
+    learning_rate: float = 1e-2
+    weight_decay: float = 0.01
+    optimizer: Literal["adam", "lbfgs"] = "adam"
+
+
+class CCS(torch.nn.Module):
     def __init__(
         self,
+        d,
         verbose=False,
         include_bias=True,
-        num_epochs=1000,
-        num_tries=10,
-        learning_rate=1e-2,
```

Comment:
Shouldn't we leave that stuff in the constructor, if want it to be like scikitlearn:
See for example:
https://scikit-learn.org/stable/modules/generated/sklearn.neural_network.MLPClassifier.html

learning_rate, batch_size etc. are set in the constructor there

Reply:
My bad, I thought sklearn API was the same as Keras' (which asks for learning params in the fit function). I think we should revert the very last commit.

Reply:
I disagree, actually. We don't have to be _exactly_ like `sklearn` and this aspect of their API doesn't really make any sense, especially for a PyTorch module. It just forces you to create a bunch of instance attributes that don't need to be there otherwise. And what if you want to fit the model multiple times with different hyperparameters?